UT4MasterServer/Controllers/Epic/PersonaController.cs

```diff
@@ -43,39 +44,59 @@ public async Task<IActionResult> AccountLookup([FromQuery(Name = "q")] string qu
 		return Json(obj);
 	}
 
+	// TODO: modify into a GET with query parameters
 	[HttpPost("accounts/search")]
 	public async Task<IActionResult> SearchAccounts([FromBody] AccountSearchRequest request)
 	{
 		if (User.Identity is not EpicUserIdentity authenticatedUser)
 		{
 			return Unauthorized();
 		}
-		var accounts = await accountService.SearchAccountsAsync(request.Query);
 
-		if (request.Roles != null && request.Roles.Length > 0)
+		// do not allow more than 50 entries
+		if (request.Take > 50)
 		{
-			accounts = accounts.Where((a) => a.Roles != null && a.Roles.Intersect(request.Roles).Any());
+			request.Take = 50;
 		}
 
-		var count = accounts.Count();
+		// TODO: duplicate code in AdminPanelController
+		AccountFlags flags;
+		if (request.Roles != null && request.Roles.Length > 0)
```

Comment:
`request.Roles?.Length` (if possible, not sure).

Reply:
Even if it is, i don't think its a good idea because it doesnt make it that clear that null check is required.

Reply:
It's clear and more modern to use conditional access